python/lsst/ci/models.py

```diff
@@ -176,3 +178,78 @@ class RepoSpec:
 
     def __str__(self):
         return self.url
+
+
+class GitHubPR(BaseModel):
+    """A single pull request for a given product/repository.
+
+    Attributes
+    ----------
+    product_name : `str`
+        The name of the product associated with the PR.
+    owner : `str`
+        The owner or organization of the repository.
+    repo : `str`
+        The repository name.
+    pr_number : `int`
+        The pull request number.
+    title : `str`
+        The title of the pull request.
+    head_ref : `str`
+        The branch name that the pull request is based on.
+    sha : `str`
+        The commit SHA associated with the pull request.
+    """
+
+    product_name: str
+    owner: str
+    repo: str
+    pr_number: int
+    title: str
+    head_ref: str
+    sha: str
+
+    def __str__(self) -> str:
+        """One liner for logging / printing."""
+        return (
+            f"PR #{self.pr_number} "
+            f"in {self.owner}/{self.repo} "
+            f"for product {self.product_name}: {self.title}"
+        )
+
+
+class AllProductPRCollections(BaseModel):
+    """All PR collections for every product.
+
+    Attributes
+    ----------
+    items : `list[GitHubPR]`
+        A list of PR collections, each representing a product
+        and its associated PRs.
+    """
+
+    items: list[GitHubPR]
+
+    def __iter__(self):
+        """Iterate directly over the contained PRs."""
+        return iter(self.items)
+
+    def __len__(self):
+        """Length of list."""
+        return len(self.items)
+
+    def __bool__(self):
+        """Empty collections are False."""
+        return bool(self.items)
+
+    def summary(self) -> None:
```

Comment:
```suggestion
    def print_summary(self) -> None:
```
Prefer this to be more active a method name. I think either `print_summary` that prints, or `summarize()` that returns a string that can be printed.